mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -182,6 +182,9 @@ static LogicalResult generateLoopNestUsingForOp(
   if (loops.empty())
     return success();
 
+  assert(tiledResults.size() == destinationTensors.size() &&
```

Comment:
I hope this is true. I am not a hundred percent sure, but I think it is true. Why do you need this assert?

Reply:
This assert is same as iter_args == number of values yielded. I think it is fair to expect that from an implementation that its tiled implementation should not generate more results than inits.